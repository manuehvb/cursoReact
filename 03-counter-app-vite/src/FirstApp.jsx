import PropTypes from 'prop-types';

const FirstApp = ({ title, subTitle }) => {


    return (
        <>
            <h1>{title}</h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subTitle}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'AAA'
}

export default FirstApp;