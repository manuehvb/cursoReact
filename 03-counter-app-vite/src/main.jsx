import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import FirstApp from './FirstApp';
import CounterApp from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title="Titulo" /> */}
        <CounterApp value={20} />
    </React.StrictMode>
);