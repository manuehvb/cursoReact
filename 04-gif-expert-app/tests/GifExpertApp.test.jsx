import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";
import { renderHook } from "@testing-library/react";

describe('Tests de <GifExpertApp />', () => {

    test('GifExpertApp snapshot', () => {
        const { container } = render(<GifExpertApp />);

        expect(container).toMatchSnapshot();
    });

    test('Debe pintar el estado inicial', () => {
        render(<GifExpertApp />);

        expect(screen.getByText('GifExpertApp')).toBeTruthy();
        expect(screen.getByRole('form')).toBeTruthy();
        expect(screen.getByRole("textbox")).toBeTruthy();
        expect(screen.getByText('Rick and Morty')).toBeTruthy();
        expect(screen.getByText('Cargando...')).toBeTruthy();
    });

});