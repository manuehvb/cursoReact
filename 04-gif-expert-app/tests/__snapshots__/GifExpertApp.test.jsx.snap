// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <GifExpertApp /> GifExpertApp snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Rick and Morty
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
